extensions/vault/runtime/src/main/java/io/quarkus/vault/internal/client/OkHttpVaultClient.java

```diff
@@ -0,0 +1,161 @@
+package io.quarkus.vault.internal.client;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+import static io.quarkus.vault.internal.client.OkHttpClientFactory.createHttpClient;
+
+import java.io.IOException;
+
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.quarkus.vault.VaultException;
+import io.quarkus.vault.config.VaultServerRuntimeConfig;
+import io.quarkus.vault.internal.client.dto.VaultAdCreds;
+import io.quarkus.vault.internal.client.dto.VaultDbCreds;
+import io.quarkus.vault.internal.client.dto.VaultKubernetesAuth;
+import io.quarkus.vault.internal.client.dto.VaultKubernetesAuthBody;
+import io.quarkus.vault.internal.client.dto.VaultKvSecretV1;
+import io.quarkus.vault.internal.client.dto.VaultKvSecretV2;
+import io.quarkus.vault.internal.client.dto.VaultLeasesBody;
+import io.quarkus.vault.internal.client.dto.VaultLeasesLookup;
+import io.quarkus.vault.internal.client.dto.VaultLookupSelf;
+import io.quarkus.vault.internal.client.dto.VaultRenewLease;
+import io.quarkus.vault.internal.client.dto.VaultRenewSelf;
+import io.quarkus.vault.internal.client.dto.VaultRenewSelfBody;
+import io.quarkus.vault.internal.client.dto.VaultUserPassAuth;
+import io.quarkus.vault.internal.client.dto.VaultUserPassAuthBody;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+public class OkHttpVaultClient implements VaultClient {
```

Comment:
We need to create a GitHub issue for migrating that to using the REST client. Let's create the issue and see how we can do it later.

Reply:
I created https://github.com/quarkusio/quarkus/issues/4162 as a feature request.